File: app/src/main/java/com/expansion/lg/kimaru/training/receivers/TremapApp.java
package com.expansion.lg.kimaru.training.receivers;

import android.app.Application;

/**
 * Created by kimaru on 2/23/18.
 */

public class TremapApp extends Application {
    public static TremapApp mInstance;
    @Override
    public void onCreate(){
        super.onCreate();
        mInstance = this;
    }

    public static synchronized TremapApp getmInstance(){
        return mInstance;
    }

    public void setConnectivityListener (ConnectivityReceiver.ConnectivityReceiverListener listener){
        ConnectivityReceiver.connectivityReceiverListener = listener;
    }
}

File: app/src/main/java/com/expansion/lg/kimaru/training/receivers/TremapApp.kt
package com.expansion.lg.kimaru.training.receivers

import android.app.Application

/**
 * Created by kimaru on 2/23/18.
 */

class TremapApp : Application() {
    override fun onCreate() {
        super.onCreate()
        mInstance = this
    }

    fun setConnectivityListener(listener: ConnectivityReceiver.ConnectivityReceiverListener) {
        ConnectivityReceiver.connectivityReceiverListener = listener
    }

    companion object {
        var mInstance: TremapApp

        @Synchronized
        fun getmInstance(): TremapApp {
            return mInstance
        }
    }
}
